static/app/views/profiling/profileDetails.tsx

```diff
@@ -149,6 +147,108 @@ function ProfileDetails() {
     [location, search]
   );
 
+  const [filters, setFilters] = useState<Partial<Record<TableColumnKey, string[]>>>({});
+
+  const columnFilters = useMemo(() => {
+    function makeOnFilterChange(key: string) {
+      return values => {
+        setFilters(prevFilters => ({
+          ...prevFilters,
+          [key]: values.length > 0 ? values.map(val => val.value) : undefined,
+        }));
+      };
+    }
+    return {
+      type: {
+        values: ['application', 'system'],
+        onChange: makeOnFilterChange('type'),
+      },
+      image: {
+        values: pluckUniqueValues(slowestFunctions, 'image').sort((a, b) =>
+          a.localeCompare(b)
+        ),
+        onChange: makeOnFilterChange('image'),
+      },
+    };
+  }, [slowestFunctions]);
+
+  const currentSort = useMemo<{key: TableColumnKey; order: 'asc' | 'desc'}>(() => {
+    const functionsSort = location.query?.functionsSort;
+    if (!functionsSort) {
+      return {
+        key: 'self weight',
+        order: 'desc',
+      };
+    }
+
+    let key = functionsSort;
+
+    const isDesc = key[0] === '-';
+    if (isDesc) {
+      key = key.slice(1);
+    }
+    return {
+      key: key as TableColumnKey,
```

Comment:
Do you need to validate that key is a valid `TableColumnKey` Since it's coming from the query params?

Reply:
Yeah that would be wise. In fact, the issue you identified earlier was due to this `functionsSort` query param sticking when returning back to the summary screen and hitting the backend with an incorrect column since i re-used the same key.

i'm pushing a fix